pig/src/main/java/org/apache/zeppelin/pig/BasePigInterpreter.java

```diff
@@ -97,4 +98,21 @@ public Scheduler getScheduler() {
   }
 
   public abstract PigServer getPigServer();
+
+  /**
+   * Use paragraph title if it exists, else use the last line of pig script.
+   * @param cmd
+   * @param context
+   * @return
+   */
+  protected String createJobName(String cmd, InterpreterContext context) {
+    String pTitle = context.getParagraphTitle();
+    if (!StringUtils.isEmpty(pTitle)) {
+      return pTitle;
+    } else {
+      // use the last line of pig script as the job name.
+      String[] lines = cmd.split("\n");
+      return lines[lines.length - 1];
```

Comment:
is it possible to end up with some unexpected characters, eg.
```
STORE same1 INTO '/tmp/tutorial-join-results' USING PigStorage();
```
like `/` or `()` or `;`?